sstables/index_reader.hh

```diff
@@ -1087,10 +1087,9 @@ public:
 
     future<> close() noexcept {
         // index_bound::close must not fail
-        return close(_lower_bound).then([this] {
-            if (_upper_bound) {
-                return close(*_upper_bound);
-            }
+        auto close_lb = close(_lower_bound);
+        auto close_ub = _upper_bound ? close(*_upper_bound) : make_ready_future<>();
+        return when_all(std::move(close_lb), std::move(close_ub)).discard_result().finally([this] {
```

Comment:
@tgrabiec / @michoecho is it possible that `_lower_bound == *_upper_bound`?
If so, I think this could lead to double close of index_bound.context since it's not reset by `close(index_bound& b)` But that could have happened also when closing serially.  closing them in parallel doesn't change that.

Reply:
I sent https://github.com/scylladb/scylladb/pull/12370 to handle that